src/NormalizingFlows.jl

```diff
@@ -1,5 +1,72 @@
 module NormalizingFlows
 
-# Write your package code here.
+using Bijectors
+using Optimisers
+using LinearAlgebra, Random, Distributions, StatsBase
+using ProgressMeter
+using ADTypes, DiffResults
+using Zygote, ForwardDiff, ReverseDiff, Enzyme
+
+using DocStringExtensions
+
+export NF, elbo, loglikelihood
+
+"""
+    NF(rng::AbstractRNG, vo, flow, args...; kwargs...)
+    NF(vo, flow, args...; kwargs...) = NF(Random.default_rng(), vo, flow, args...; kwargs...)
+
+Train the given normalizing flow `flow` by calling `train`.
+
+# Arguments
+- `rng::AbstractRNG`: random number generator
+- `vo`: variational objective
+- `flow`: normalizing flow to be trained
+- `args...`: additional arguments for `vo`
+- `max_iters::Int=1000`: maximum number of iterations
+- `optimiser::Optimisers.AbstractRule=Optimisers.ADAM()`: optimiser to compute the steps
+- `ADbackend::ADTypes.AbstractADType=ADTypes.AutoZygote()`: automatic differentiation backend
+- `kwargs...`: additional keyword arguments for `train` (See `train`)
+
+# Returns
+- `flow_trained`: trained normalizing flow
+- `opt_stats`: statistics of the optimiser during the training process (See `train`)
+- `st`: optimiser state for potential continuation of training
+"""
+function NF end
+
+NF(vo, flow, args...; kwargs...) = NF(Random.default_rng(), vo, flow, args...; kwargs...)
+function NF(
+    rng::AbstractRNG,
+    vo,                                      # elbo, likelihood, f-div, STL, etc.. (how do we deal with this? it would require different input)
+    flow,                                    # normalizing flow to be trained
+    args...;                                 # additional arguments for vo 
+    max_iters::Int=1000,
+    optimiser::Optimisers.AbstractRule=Optimisers.ADAM(),
+    ADbackend::ADTypes.AbstractADType=ADTypes.AutoZygote(),
+    kwargs...,
+)
+    # destruct flow for explicit access to the parameters
+    @info "destructuring flow object"
```

Comment:
I know this was done because you ran into the issue of `destructure` taking ages, but it's also "noisy"; it's not particularly interesting to see this message in my terminal every time I run `NF`. 

At the very least maybe add a `verbose=false` keyword, and make this line
```suggestion
    verbose && @info "destructuring flow object"
```

Reply:
We should also probably inform the user that "Hey! You FunctionChains if you have a deep network; don't use composition!"

Reply:
I'll probably just remove this one---I agree that this is noisy and not informative.